src/hardware/node_info.hpp

```diff
@@ -7,14 +7,16 @@ namespace hw {
 // Information about the computational resources available on a compute node.
 // Currently a simple enumeration of the number of cpu cores and gpus, which
 // will become richer.
-struct node {
-    node();
-    node(int c, int g);
+struct node_info {
+    node_info() = default;
+    node_info(unsigned c, unsigned g);
```

Comment:
I don't think this constructor adds anything that we wouldn't get just by simple aggregate initialization, but if it is kept we could just have the ctor body inline here and save any concern about a potentially complicated implementation that would warrant its source be placed in a separate file. The explicit default constructor could also be removed if the two-parameter ctor is removed.